File: 品味Java的21種設計模式/dp_cpp/abstractfactory/example5/XmlDAOFactory.cpp
#include "XmlDAOFactory.h"

namespace cn {
namespace javass {
namespace dp {
namespace abstractfactory {
namespace example5 {

OrderDetailDAO *XmlDAOFactory::createOrderDetailDAO() {
    return new XmlDetailDAOImpl();
}

OrderMainDAO *XmlDAOFactory::createOrderMainDAO() {
    return new XmlMainDAOImpl();
}
}
}
}
}
}

File: 品味Java的21種設計模式/dp_java/abstractfactory/example5/XmlDAOFactory.java
package cn.javass.dp.abstractfactory.example5;

public class XmlDAOFactory extends DAOFactory {
    public OrderDetailDAO createOrderDetailDAO() {
        return new XmlDetailDAOImpl();
    }
    public OrderMainDAO createOrderMainDAO() {
        return new XmlMainDAOImpl();
    }
}